lib/mongoid/lazy_migration/document.rb

```diff
@@ -63,8 +63,7 @@ def try_lock_migration
     selector = atomic_selector.merge('migration_state' => { "$in" => [nil, 'pending'] })
     changes  = { "$set" => { 'migration_state' => 'processing' }}
     safety   = { :safe => true }
-
-    self.class.with(safety).where(selector).query.update(changes)['updatedExisting']
+    self.class.with(safety).where(selector).update(changes).successful?
```

Comment:
I think you should read through the persistence document above. There are apparently some potential issues with `with` as well which can leave database connections open...

| Also note another important gotcha that could result in the number of connections continually growing with each use of #with. Because a new client is created under the covers in #with, any options that would cause a driver cluster configuration change will instantiate a client that would never get garbage collected due to the new cluster’s monitor threads. There are two ways to avoid opening up new connections via the driver that are never closed.